src/Analyzer/Utils.h

```diff
@@ -160,4 +160,14 @@ QueryTreeNodePtr buildSubqueryToReadColumnsFromTableExpression(const NamesAndTyp
 QueryTreeNodePtr buildSubqueryToReadColumnsFromTableExpression(const QueryTreeNodePtr & table_node, const ContextPtr & context);
 
 
+bool hasUnknownColumn(
+    const QueryTreeNodePtr & node,
+    QueryTreeNodePtr replacement_table_expression);
+
+void replaceFilterExpression(
+    QueryTreeNodePtr & expression,
+    const QueryTreeNodePtr & replacement_table_expression,
+    const ContextPtr & context);
```

Comment:
Nitpick: better naming

```suggestion
/// Removes filter expressions that reference columns from table expressions other than `allowed_table_expression`
void removeUnknownFilterExpressions(
    QueryTreeNodePtr & expression,
    const QueryTreeNodePtr & allowed_table_expression,
    const ContextPtr & context);
```